index.js

```diff
@@ -1,44 +1,149 @@
 // import your node modules
-const express = require('express')
-
-const db = require('./data/db.js');
+const express = require('express');
+const db = require('./data/db');
 
 // add your server code starting here
 const server = express();
+
 const PORT = 4545;
 
+//++++++++++++++++++++++++ 
+// middlewear  and addins
+//+++++++++++++++++++++++
+const parser = express.json();
+server.use(parser);
+
+
+// wire up middleware
+// server.use(express.json());
+
+
+
 server.get('/api/posts', (req, res) => {
     db.find()
-    .then(posts => {
-            console.log(`posts ${posts}`);
-            res.status(200).send(posts);
-        
-    })
-    .catch(err =>{
-        res.status(500).send(`<h1>Bad Request<h1>`);
-        //res.status(500).json({message: `failed to get users`});
+    .then( posts => {
+            console.log(`posts = ${posts}`);
+            res.status(200).send(posts); 
     })
+    .catch(err => {
+        // res.status(500).send(`<h1>Bad Request<h1>`);
+        res.status(500).json({message: `failed to get users`});
+        //res.(status(500).json(err);
+    });
 });
     
 
-server.get('/api/posts/:id', (req,res) => {
+server.get('/api/posts/:id', (req, res) => {
     // console.log(req);
     console.log(req.params);
-    const thisId = req.params.id;
-    db.findById(thisId)
-        .then( thisPost => {
-            console.log(`thisPost = ${thisPost}`);
-            res.status(200).send(thisPost);
+    const id = req.params.id;
+
+    db.findById(id)
+        .then( posts => {
+            console.log(`posts = ${posts.id}`);
```

Comment:
This console.log seems a little confusing (saying `posts equals [some post id]`)